segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/ConcurrencyManager.java

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.impl.bookkeeper;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import io.pravega.common.AbstractTimer;
+import io.pravega.common.MathHelpers;
+import java.util.function.Supplier;
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.ThreadSafe;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Manages the Write Queue Concurrency.
+ */
+@ThreadSafe
+class ConcurrencyManager {
+    //region Members
+
+    @VisibleForTesting
+    static final double SIGNIFICANT_DIFFERENCE = 0.25;
+    @VisibleForTesting
+    static final long UPDATE_FREQUENCY_MILLIS = 1000;
```

Comment:
Let's make this less aggressive. For example, let's use at least 5 seconds. Also, let's be precise and call it period, not frequency. Frequency is number of events over time, whereas this is time only.

Reply:
fixed